hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/Job.groovy

```diff
@@ -48,6 +48,7 @@ class Job {
   Map<String, Object> jobProperties;
   Map<String, String> reading;
   Map<String, String> writing;
+  HashSet<String> requiredParameters;
```

Comment:
Could you change the declaration of requiredParameters to a Set<String>, but in the constructor, create it as a LinkedHashSet<String>? Using a LinkedHashSet instead of a regular HashSet will preserve the ordering of the requiredParameters (when enumerated). This will make it possible for us to have consistent unit test outputs.